### PR TITLE
Fix issue with compiler vs swift

### PR DESCRIPTION
**Pull request by minuscorp**
Fixes an issue with the `swift` vs `compiler` flag.

### CODE REVIEW

**Comment by orta** [comment]
Is it worth dropping those logs?

File: Sources/PackageConfig/Package.swift [hunk 1]
```diff
@@ -163,24 +163,26 @@ enum Package {
         // /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/4_2
         let fileManager = FileManager.default
         let swiftPMDir = swiftPath.replacingOccurrences(of: "bin/swiftc", with: "lib/swift/pm")
-        #if swift(<5.5.2)
+        let swiftToolsVersion = getSwiftToolsVersion()
+        #if compiler(>=5.5.2)
+        debugLog("MORE 5.5.2")
```
Comment (minuscorp) [inline Sources/PackageConfig/Package.swift:168]:
```suggestion
```

File: Sources/PackageConfig/Package.swift [hunk 2]
```diff
@@ -163,24 +163,26 @@ enum Package {
...
+        // Since Swift 5.5.2 there're no more different versions installed in the toolchain and it is stored
+        // in another directory
+        let directory = try! fileManager.contentsOfDirectory(atPath: swiftPMDir)
+            .first(where: { $0.starts(with: "Manifest") })
+        let spmManifestDir = directory!
+        #else
+        debugLog("LESS 5.5.2")
```
Comment (minuscorp) [inline Sources/PackageConfig/Package.swift:175]:
```suggestion
```

**Comment by minuscorp** [comment]
Better without them yeah

**Comment by orta** [comment]
👍🏻

**Pull request merged**